### PR TITLE
InteractiveTutorials Prototyping Blog Post

### PR DESCRIPTION
**Pull request by madeleinesinging**
I created this blog post as part of my 2022 Intern Project to help prototype the InteractiveTutorials gem. 

After attending other intern final project demos on Chime, I used the same general style as they did, but I'm not sure if the tone or the structure should be changed since this is a public blog post. I'm also not sure if the pictures I added are appropriate or if I went into too much/too little detail about certain aspects of my work. Any comments/criticism is appreciated! 

image
image
image
image

### CODE REVIEW

File: content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md
Comment (cgalvan) [inline content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md:10]:
```suggestion
Hi! I’m Madeleine, an Amazon intern working on Open 3D Engine (O3DE). My summer project focused on advanced prototyping of the InteractiveTutorials gem. So far, I've added several features and scoped out prospective future improvements. I also worked on automating and making tutorials easier to navigate.
```

I think adding `also` here makes the distinction that this is in addition to the features you added and scoping out of future improvements.

File: content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md
Comment (cgalvan) [inline content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md:17]:
```suggestion
From there, I began to work on the tutorials themselves. Keeping with the Rigid Bodies tutorial's theme of PhysX, I wrote tutorials describing how to create wind forces, process PhysX collider assets, decompose input meshes, and customize mesh assets. These tutorials already existed on the O3DE website, but having minimal previous experience with documentation writing, especially for game engines, I wasn't sure exactly how much information to include in one step, how much background should be given about the general function of different components and assets, or even what tense to use. It turns out that one of the best ways to learn is by doing: trying and failing and mimicking those who already know. After lots of edits, my wind forces tutorial was added to the gem.
```

File: content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md
Comment (cgalvan) [inline content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md:33]: I would break this part into a new paragraph and mention that we had to make an RFC and get it approved, which ties in with your final sentence about the importance of communication, since we had to talk with multiple teams/stakeholders.

File: content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md
Comment (cgalvan) [inline content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md:40]:
```suggestion
For future work on automation, I would suggest working bottom-up rather than top-down. I would attempt to automate a very simple tutorial, such as Create an Entity, and to develop a general purpose solution, rather than jumping out of the gate and trying to automate the entire Rigid Bodies tutorial on a first pass. I would also consider using an abstract data type to memorize or keep track of the current-step-index and the last-simulated-step index, since many problems which rely on consecutive if-else gates can also be solved dynamically. Finally, instead of keeping track of the last-simulated-step-index, I might consider using some of the Python Tests in O3DE to determine which entities have already appeared and been edited. If a test returns False, then the entities it tracks must not have been created yet. An advantage of O3DE being open source is that I can continue to learn and be curious about this project long after the summer’s over. This may sound trite, but I've genuinely enjoyed my summer at Amazon, and I'm so thankful not only for the opportunity to write here, but also to have been part of a team that encouraged and challenged me to work with rigor and attention to detail.
```

**Review by cgalvan — APPROVED** [review]
A couple of minor notes, but overall this is awesome, great job! :)

**Comment by cgalvan** [comment]
General question for the docs folks, is there a way to see the blog posts from the o3de.org home page? I was only able to find it after searching "blog" in the search bar and then found this link:

`https://deploy-preview-1894--o3deorg.netlify.app/blog/`

**Comment by micronAMZN** [comment]
> General question for the docs folks, is there a way to see the blog posts from the o3de.org home page? I was only able to find it after searching "blog" in the search bar and then found this link:

This is the blog index page from the deploy from this PR: 

https://deploy-preview-1894--o3deorg.netlify.app/community/

The link shows how the site would look to anyone who clicks on the community link if this blog was merged.

File: content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md
Comment (micronAMZN) [inline content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md:26]: This image is very large and when I suggested to add it, I had not seen the blog post in this format with the image at the top of the article. Apologies but I think this image should be removed because the top image is much better and serves the article.

```suggestion
```

File: content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md
Comment (micronAMZN) [inline content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md:38]: typo?

```suggestion
I risked creating duplicate entities or potentially breaking the tutorial by trying to edit entities which didn't yet exist when the 'simulate' option was selected. I knew I could solve the problem by condensing the tutorial into three giant steps, but that seemed inefficient and also unhelpful for the user. I decided to create a 'simulate last step' option which displayed the completed result of the tutorial as a reference for the user.
```

**Review by micronAMZN — APPROVED** [review]
Approving with two minor suggestions.

**Comment by cgalvan** [comment]
> 

Ah awesome, thanks!

**Comment by cgalvan** [comment]
> > General question for the docs folks, is there a way to see the blog posts from the o3de.org home page? I was only able to find it after searching "blog" in the search bar and then found this link:
> 
> This is the blog index page from the deploy from this PR:
> 
> https://deploy-preview-1894--o3deorg.netlify.app/community/
> 
> The link shows how the site would look to anyone who clicks on the community link if this blog was merged.

Ah awesome, thanks!

File: content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md
Comment (aFinchy) [inline content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md:4]: Can we please change the date to September the 7th?

File: content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md
Comment (aFinchy) [inline content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md:5]: Might I suggest adding a proper Slug? Maybe
"interactive-tutorials-prototyping"

**Comment by aFinchy** [comment]
Here is how the blog currently looks like. 
image

File: content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md
Comment (aFinchy) [inline content/blog/posts/o3de-interactive-tutorials-prototyping-2022.md:7]: I would suggest using a different image for this one as its getting messed up on the blog list. 
dev-series112

**Review by aFinchy — CHANGES_REQUESTED** [review]
Left a few changes that would need to be fixed before merging. Other than that everything looks good.

**Review by aFinchy — APPROVED** [review]
Built on Desktop and everything looks fine on my end. Approving this blog.

**Pull request merged**
